migrate/migrate_repos.py

```diff
@@ -0,0 +1,74 @@
+import itertools
+import os
+import sys
+import time
+
+import click
+from ghapi.all import GhApi, paged
+
+@click.command()
+@click.argument('src_org')
+@click.argument('dest_org')
+@click.argument('repo_list_file', type=click.File('r'))
+@click.option(
+    '--preview',
+    is_flag=True,
+    help="Preview what will happen but don't execute."
+)
+@click.option(
+    '--skip-missing',
+    is_flag=True,
+    help="Skip repos that are not found in the source org, instead of failing. Useful after an error."
+)
+def migrate(src_org, dest_org, repo_list_file, preview, skip_missing):
+    if preview:
+        click.echo("In Preview Mode: No changes will be made!")
+
+    if src_org == dest_org:
+        sys.exit("Fatal Error: Source and destination orgs must be different.")
+
+    github_token = os.environ.get('GITHUB_TOKEN')
+    if not github_token:
+        sys.exit("Fatal Error: Please set a GITHUB_TOKEN environment variable.")
+
+    # Get the list of repos from our text file.
+    repos_to_transfer = [
+        repo_name.strip() for repo_name in repo_list_file if repo_name.strip()
+    ]
+    click.echo(f"Read {len(repos_to_transfer)} repos from {repo_list_file.name}")
+
+    api = GhApi(token=github_token)
+    # Basic sanity check to make sure the repos we're transferring all actually exist.
+    click.echo(f"Fetching the list of repos in source org {src_org}...")
+    src_org_repos = {
+        repo['name']
+        for repo
+        in itertools.chain.from_iterable(paged(api.repos.list_for_org, src_org, per_page=100))
```

Comment:
I like the itertools.chain.from_iterable here, but surprised that `paged` doesn't do that for you :(

Reply:
Yeah, I was a little surprised at first too. But thinking on it more, I actually kind of prefer that it doesn't. The `paged` function is a convenience, to be sure, but it never actually lets you forget that you're making a separate HTTP call for each page you iterate through. If a client (like me in this case) decides they _want_ to ignore that fact because they know it's only ever going to be at most three pages, then they can always break out `itertools` like above and smash it together.